android/src/main/java/com/ryanheise/audioservice/AudioService.java

```diff
@@ -584,6 +602,8 @@ private void play(Runnable runner) {
 			registerNoisyReceiver();
 			mediaSession.setSessionActivity(contentIntent);
 			startForeground(NOTIFICATION_ID, buildNotification());
+			if (androidStopForegroundOnPause)
+				isForeground = true;
```

Comment:
There appears to be an API to test whether a service is in the foreground, but it's post API level 29, so this solution is fine. Although I'd delete the condition and just have this field store whether or not the service is in the foreground. Even though it's only used in this particular use case, the meaning of the field would become much more easy to reason about.

Reply:
Yes the condition can be removed.